src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -0,0 +1,338 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.DotNet.XHarness.Android
+{
+    public class AdbRunner
+    {
+        #region Constructor and state variables
+
+        private readonly string _absoluteAdbExePath;
+        private readonly ILogger _log;
+
+        public AdbRunner(ILogger log, string adbExePath = ".\\adb.exe")
+        {
+            _log = log;
+
+            // We need to find ADB.exe somewhere
+            string environmentPath = Environment.GetEnvironmentVariable("ADB_EXE_PATH");

```

Comment:
Could use a constant for ADB_EXE_PATH, later you can log the usage using the constant, I tend to do typos, so the less I write the less typos :)

Reply:
Dione